mlir/test/Quantum/CanonicalizationTest.mlir

```diff
@@ -21,10 +21,10 @@ func.func @test_alloc_dce() {
     return
 }
 
-// CHECK-LABEL: test_alloc_no_cse
-func.func @test_alloc_no_cse() -> (!quantum.reg, !quantum.reg){
-    // CHECK: quantum.alloc
+// CHECK-LABEL: test_alloc_cse
+func.func @test_alloc_cse() -> (!quantum.reg, !quantum.reg){
     // CHECK: quantum.alloc
+    // CHECK-NOT: quantum.alloc
     %r1 = quantum.alloc(4) : !quantum.reg
     %r2 = quantum.alloc(4) : !quantum.reg
     return %r1, %r2 : !quantum.reg, !quantum.reg
```

Comment:
We should make clear that this is a regression that might need fixing in the future (if we ever support multiple registers). Several places would be revenant: in the changelog, the PR description (drawbacks), and somewhere in the code.

Reply:
Added in the .td file as well.